src/qml/controls/IPAddressValueInput.qml

```diff
@@ -16,7 +16,7 @@ TextInput {
     property bool validInput: true
     enabled: true
     state: root.parentState
-    validator: RegExpValidator { regExp: /[0-9.:]*/ } // Allow only digits, dots, and colons
+    validator: RegExpValidator { regExp: /[0-9A-Fa-f.:\[\]]*/ } // Allow only IPv4/ IPv6 chars
```

Comment:
I read https://doc.qt.io/qt-5/qml-qtquick-regexpvalidator.html but it is unclear whether the framework would prefix the regexp with `^` and suffix it with `$`. If it does not then we need to add it ourselves, because `[0-9.:]*` matches anything, e.g.:

```sh
$ echo xyz |grep -E '[0-9.:]*'
xyz
```
(this is because by default there is a match if the input _contains_ a matching pattern and `[0-9.:]*` denotes 0 or more chars, so "xyz" contains 0 of those chars followed by "xyz")

In the examples from the doc they add an explicit `$` at the end but not `^` at the front, hmm?

In a bracket expression `\` looses its special meaning. So the above closes the bracket expression at the first `]`: `[0-9A-Fa-f.:\[\]` followed by a literal `]`. Probably the regex should be something like `^[][0-9A-Fa-f.:]*$`, but is this even needed, given that there is a validation function?

Reply:
> but is this even needed, given that there is a validation function?

It's a first landing and it will avoid the user to enter invalid symbols or chars, intentionally or not.

Reply:
> In a bracket expression \ looses its special meaning... Probably the regex should be something like ^[][0-9A-Fa-f.:]*$

That's true for the terminal, but it seems regular expressions are handled a bit differently in QML.

In the terminal, `grep` interprets `[` and `]` differently and doesn’t use the same escaping rules as some programming environments (like QML). When using grep, we need to adjust the syntax to avoid issues with special characters and escaping but QML uses JavaScript-based regex syntax (similar to ECMAScript). I can use your suggestion as it works in both environments and the one I was adding was only working in QML.

Anyways, the idea of having this regex validator (in fact looking at the doc ref you provided above I also realised that class is deprecated and should use `RegularExpressionValidator` now) is intended to ensure that an input contains only characters that are valid in IPv4 and IPv6 addresses, not to validate the format (as it would do the logic added into the node interface), it's purely esthetic. I've tried other alternatives like `inputMask:` (horrible UX) and classes like `QHostAddress` or `QNetworkAddressEntry` but the regex validator seems very practical and simplest.

Reply:
After playing with this a bit and reading https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_class#non-v-mode_character_class I think it should be:

```sh
/^[\][0-9a-f.:]*$/i
```

```js
#!/usr/bin/env node

if ("[A]:2.3b".match(/^[\][0-9a-f.:]*$/i)) {
    console.log("match");
} else {
    console.log("no match");
}
```

Do we want to accept the empty string as valid? If no, then change the `*` to `+`.

Reply:
Ok, let me check that again. Thanks!

Reply:
As commented to you offline, current implementation of the regex works as expected and doesn't allow spaces, it also works on a terminal with grep.

But I tried both current and your suggestion on https://jsconsole.com/, current implementation doesn't work well there but your does so I'm deciding to use yours (which includes `i` flag that allows case-insensitive); regarding `*` and `+`, it doesn't matter as the validation is being triggered anyways but for consistency and since it works in the `jsconsole`, I'll use `+` too.